pandas/core/window.py

```diff
@@ -788,6 +789,10 @@ def __init__(self, obj, *args, **kwargs):
     corr = GroupByMixin._dispatch('corr', other=None, pairwise=None)
     cov = GroupByMixin._dispatch('cov', other=None, pairwise=None)
 
+    # Removed from type checking because there is currently no good way to
+    # handle multiple inheritance with mypy
+    # https://github.com/python/mypy/issues/2125
+    @no_type_check
```

Comment:
What is the motivation to use `no_type_check` instead of `# type: ignore`? There is some overlap but I think the latter is what we should be using when we want to explicitly ignore errors generated by the static analysis

Reply:
`type: ignore` placed on the either or both lines defining `_apply` in the mixins doesn't fix the problem. Placing it on the line before the `def` at the same indent level would probably work, but it will have the undesired side effect of turning off type checking at that indent level for the rest of the class. And there's no place in the inheriting classes to place an `ignore` because those classes don't override `_apply`.

As I understand [PEP 484](https://www.python.org/dev/peps/pep-0484/), `@no_type_check` is designed to ignore a whole class or function, which is what we want here.

Reply:
Wouldn't we still want type checking within the function though?

Reply:
That would be ideal, but I haven't been able to figure out a way to ignore the signature mismatch but not the body of the function. Open to suggestions.

Reply:
Just adding `# type: ignore` at the end of the signature doesn't do it?

Reply:
Unfortunately not.

Reply:
We can always bring the two signatures in line with one another. `_GroupByMixin._apply` is only different from `_Rolling._apply` by a default value for the name argument (no default in _GroupByMixin and `None` as the default in `_Rolling`.

Reply:
Yea I think aligning works just as well; these are all internal methods